flax/core/scope.py

```diff
@@ -323,9 +324,9 @@ def reserve(self, name: str):
       name: the name to reserve.
     """
     if not isinstance(name, str):
-      raise ValueError('Variable and child scopes should have a string name.')
+      raise ScopeNamingError('{name} is not a string.')
```

Comment:
Would the "mini-guide" for this use of `ScopeNamingError` and the one right below be different? (I would think so -- the thing you'd have to do to fix it, or what you need to understand behind it are different) so should they be different class types?

Reply:
Yes I agree, updated it.